script/test

```diff
@@ -239,23 +244,23 @@ do_expect()
 {
     local ot_command
 
-    if [[ "${NODE_MODE}" == rcp ]]; then
-        ot_command="${OT_CLI_PATH} ${RADIO_DEVICE}"
+    if [[ ${NODE_MODE} == rcp ]]; then
+        ot_command="$OT_CLI_PATH $RADIO_DEVICE"
     else
-        ot_command="${OT_BUILDDIR}/cmake/openthread-simulation-${THREAD_VERSION}/examples/apps/cli/ot-cli-ftd"
+        ot_command="$OT_BUILDDIR/cmake/openthread-simulation-$THREAD_VERSION/examples/apps/cli/ot-cli-ftd"
     fi
 
     while read -r script; do
         rm -rf tmp
-        OT_COMMAND="${ot_command}" "${script}" || {
+        OT_COMMAND="$ot_command" "$script" || {
             local exit_code=$?
-            echo -e "${COLOR_FAIL}FAIL${COLOR_NONE} ${script}"
-            exit "${exit_code}"
+            echo -e "${COLOR_FAIL}FAIL$COLOR_NONE $script"
```

Comment:
prefer `${var}` over `$var` from https://google.github.io/styleguide/shellguide.html#variable-expansion

Reply:
`shellharden` seems to want this change. Unless you can suggest how to configure `shellharden` otherwise, we can either:

1. Remove `shellharden` from `script/make-pretty`

2. Accept this style.

Thoughts?

Reply:
Since bash is not easy to debug, so I prefer the curly style which is less error prone.

I tested with the following script, no warnings are given.

```bash
a=1
ab=1
a_b=1
echo "$a_b"
echo "$a"
echo "$ab"
```

Thoughts?